### ISSUE TITLE
Refactor API and configs to better support SSL pretraining and finetuning

### ISSUE THREAD

**Issue by tigist-d** [FlyreelAI/sesemi#35]
**Is your feature request related to a problem? Please describe.**
The API used for the classifier can be simplified to better support self-supervised pretraining and supervised finetuning for semi-supervised learning.

**Describe the solution you'd like**
Selectively enable the supervised head and loss.
